cpp/arcticdb/pipeline/write_frame.cpp

```diff
@@ -130,19 +131,18 @@ folly::Future<entity::VariantKey> write_multi_index(
     return writer.commit();
 }
 
-std::vector<folly::Future<SliceAndKey>> slice_and_write(
+folly::Future<std::vector<SliceAndKey>> slice_and_write(
         InputTensorFrame &frame,
         const SlicingPolicy &slicing,
         folly::Function<stream::StreamSink::PartialKey(const FrameSlice &)>&& partial_key_gen,
         const std::shared_ptr<stream::StreamSink>& sink,
         const std::shared_ptr<DeDupMap>& de_dup_map,
         bool sparsify_floats) {
     ARCTICDB_SUBSAMPLE_DEFAULT(SliceFrame)
-    auto slices = slice(frame, slicing);
+    auto slices = std::make_shared<std::vector<FrameSlice>>(slice(frame, slicing));
```

Comment:
Why is this now a `shared_ptr`

Reply:
because it is created in this function and used later by the actual threads, after the batch_write operation is done, as part of the last ThenValue clause inside write_slices. If we don't convert it as a shared pointer, at the time it will be used, the object will be already destroyed.

Reply:
done